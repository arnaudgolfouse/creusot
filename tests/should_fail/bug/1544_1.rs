extern crate creusot_contracts;
use creusot_contracts::{logic::ops::AddLogic, prelude::*};

pub fn should_not_ice() {
    let x: S<42> = S;
    let _ = snapshot!(x + x);
}

struct S<const X: usize>;
impl<const X: usize> AddLogic for S<X> {
    // Missing the actual items! Caused an ICE in the termination check
}

pub struct BaseCurrency<I, const D: usize>([I; D]);

// Caused an ICE when computing the specialization graph
impl<I, const D: usize> PartialEq for BaseCurrency<I, D> {}
