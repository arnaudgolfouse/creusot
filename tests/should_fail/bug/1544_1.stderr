error[E0046]: not all trait items implemented, missing: `Output`, `add`
  --> 1544_1.rs:10:1
   |
10 | impl<const X: usize> AddLogic for S<X> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `Output`, `add` in implementation
   |
   = help: implement the missing item: `type Output = /* Type */;`
   = help: implement the missing item: `fn add(self, _: S<X>) -> <Self as creusot_contracts::logic::ops::AddLogic<S<X>>>::Output { todo!() }`

error[E0046]: not all trait items implemented, missing: `eq`
  --> 1544_1.rs:17:1
   |
17 | impl<I, const D: usize> PartialEq for BaseCurrency<I, D> {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `eq` in implementation
   |
   = help: implement the missing item: `fn eq(&self, _: &BaseCurrency<I, D>) -> bool { todo!() }`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0046`.
